### ISSUE TITLE
Support for the "(" class

### ISSUE THREAD

**Issue by billdenney** [eddelbuettel/digest#128]
Within certain formula, the "(" class is used for parenthetical notation.  I think that it's mainly (or perhaps only) when the parenthesis is the first item in the formula.  Regardless, the fix is pretty simple.  (I learn new things about R all the time; I had assumed that formula were made only of names and calls.)

``` r
library(digest)
sha1(a~(c+d))
#> Error: sha1() has no method for the '(' class
class((a~(c+d))[[3]])
#> [1] "("
"sha1.(" <- digest:::sha1.formula
sha1(a~(c+d))
#> [1] "3975451db3604aa3f8a947af38f29ed7d3930cb3"
```

<sup>Created on 2019-11-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by eddelbuettel** [comment]
> I learn new things about R all the time.

That makes it two of us :)  Can you can example?

**Comment by billdenney** [comment]
The example that I found was:

`sha1(a~(c+d))`

(See above for the full reproducible example.)

For the “teachable moment”, was the question relating to missing a real-world example of it?  (As in, was I trying to fix a problem that doesn’t really exist?)  I hit this bug trying to apply the updated `sha1()` on formula to my PKNCA package.

**Comment by eddelbuettel** [comment]
Ok, that is close to voodoo for me :)  I let @ThierryO chime in.

**Comment by ThierryO** [comment]
Can you provide an example where the problem occurs and the parenthese make sense? IMHO you should write your example as `sha1(a ~ c + d)`. Parenthesis at the start of the formula are not the issue. E.g. `sha1(a ~ (c + d)*e)` works.

**Comment by billdenney** [comment]
The way that I found this issue was an automatically-generated formula that has the non-simplified notation like the one above (`treatmentX ~ (time | treatment + ID)` is the actual trigger).  It exists this way because of how `update.formula()` creates the new formula.

The way that I created this formula within the PKNCA package was as follows:

```r
tmp.dose.data <- data.frame(treatment=1, ID=1)
tmp_formula <-
  stats::as.formula(
    paste0(
      ".~.|",
      paste(names(tmp.dose.data), collapse="+")
    )
  )
col.dose <- paste0(max(names(tmp.dose.data)), "X")
tmp_formula_new <-
  stats::update.formula(tmp_formula, paste0(col.dose, "~."))
sha1(tmp_formula_new)
```

`tmp_formula_new` has the improper notation with extra parentheses from the way that `update.formula()` works.

While a human wouldn't often write the formula this way, base R sometimes does.
